diff -Naur snort-2.9.1/src/dynamic-examples/dynamic-preprocessor/Makefile.am snort-2.9.1.dlucio/src/dynamic-examples/dynamic-preprocessor/Makefile.am
--- snort-2.9.1/src/dynamic-examples/dynamic-preprocessor/Makefile.am	2009-10-19 16:46:58.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-examples/dynamic-preprocessor/Makefile.am	2011-08-24 12:49:57.515573888 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include
 
-noinst_libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+noinst_libdir = ${libexecdir}/dynamicpreprocessor
 
 noinst_lib_LTLIBRARIES = lib_sfdynamic_preprocessor_example.la
 
-lib_sfdynamic_preprocessor_example_la_LDFLAGS = -export-dynamic
+lib_sfdynamic_preprocessor_example_la_LDFLAGS = -export-dynamic -module -avoid-version
 
 BUILT_SOURCES = \
 sf_dynamic_preproc_lib.c  \
diff -Naur snort-2.9.1/src/dynamic-examples/dynamic-rule/Makefile.am snort-2.9.1.dlucio/src/dynamic-examples/dynamic-rule/Makefile.am
--- snort-2.9.1/src/dynamic-examples/dynamic-rule/Makefile.am	2011-06-07 19:33:10.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-examples/dynamic-rule/Makefile.am	2011-08-24 12:52:04.876981654 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include
 
-noinst_libdir = ${exec_prefix}/lib/snort_dynamicrules
+noinst_libdir = ${libexecdir}/dynamicrules
 
 noinst_lib_LTLIBRARIES = lib_sfdynamic_example_rule.la
 
-lib_sfdynamic_example_rule_la_LDFLAGS = -export-dynamic @XCCFLAGS@
+lib_sfdynamic_example_rule_la_LDFLAGS = -export-dynamic @XCCFLAGS@ -module -avoid-version
 
 BUILT_SOURCES = \
 sfsnort_dynamic_detection_lib.c \
diff -Naur snort-2.9.1/src/dynamic-plugins/sf_engine/Makefile.am snort-2.9.1.dlucio/src/dynamic-plugins/sf_engine/Makefile.am
--- snort-2.9.1/src/dynamic-plugins/sf_engine/Makefile.am	2011-06-07 19:33:11.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-plugins/sf_engine/Makefile.am	2011-08-24 13:04:53.554371863 -0500
@@ -5,9 +5,9 @@
 
 lib_LTLIBRARIES = libsf_engine.la
 
-libdir = ${exec_prefix}/lib/snort_dynamicengine
+libdir = ${libexecdir}/dynamicengine
 
-libsf_engine_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_engine_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version -avoid-version
 
 BUILT_SOURCES = \
 sfhashfcn.c \
diff -Naur snort-2.9.1/src/dynamic-preprocessors/dcerpc2/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/dcerpc2/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/dcerpc2/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/dcerpc2/Makefile.am	2011-08-24 13:03:47.323199868 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_dce2_preproc.la
 
-libsf_dce2_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_dce2_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_dce2_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/dns/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/dns/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/dns/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/dns/Makefile.am	2011-08-24 13:04:07.968941761 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_dns_preproc.la
 
-libsf_dns_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_dns_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_dns_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/ftptelnet/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/ftptelnet/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/ftptelnet/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/ftptelnet/Makefile.am	2011-08-24 13:03:41.334274739 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_ftptelnet_preproc.la
 
-libsf_ftptelnet_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_ftptelnet_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_ftptelnet_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/imap/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/imap/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/imap/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/imap/Makefile.am	2011-08-24 13:04:01.333024720 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_imap_preproc.la
 
-libsf_imap_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_imap_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_imap_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/pop/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/pop/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/pop/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/pop/Makefile.am	2011-08-24 13:04:44.563484266 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_pop_preproc.la
 
-libsf_pop_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_pop_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_pop_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/reputation/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/reputation/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/reputation/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/reputation/Makefile.am	2011-08-24 13:04:22.847755749 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_reputation_preproc.la
 
-libsf_reputation_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_reputation_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_reputation_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/rzb_saac/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/rzb_saac/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/rzb_saac/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/rzb_saac/Makefile.am	2011-08-24 13:04:40.223538522 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-pkglibdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+pkglibdir = ${libexecdir}/dynamicpreprocessor
 
 pkglib_LTLIBRARIES = sf_rzb_saac_preproc.la
 
-sf_rzb_saac_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+sf_rzb_saac_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 PREPROCLIB=../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/sdf/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/sdf/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/sdf/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/sdf/Makefile.am	2011-08-24 13:04:17.696820145 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_sdf_preproc.la
 
-libsf_sdf_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_sdf_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_sdf_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/sip/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/sip/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/sip/Makefile.am	2011-07-14 13:35:10.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/sip/Makefile.am	2011-08-24 13:04:13.064878053 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_sip_preproc.la
 
-libsf_sip_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_sip_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_sip_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/smtp/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/smtp/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/smtp/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/smtp/Makefile.am	2011-08-24 13:03:34.381361664 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_smtp_preproc.la
 
-libsf_smtp_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_smtp_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_smtp_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/ssh/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/ssh/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/ssh/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/ssh/Makefile.am	2011-08-24 13:04:35.625596005 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_ssh_preproc.la
 
-libsf_ssh_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_ssh_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_ssh_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
diff -Naur snort-2.9.1/src/dynamic-preprocessors/ssl/Makefile.am snort-2.9.1.dlucio/src/dynamic-preprocessors/ssl/Makefile.am
--- snort-2.9.1/src/dynamic-preprocessors/ssl/Makefile.am	2011-07-13 17:44:51.000000000 -0500
+++ snort-2.9.1.dlucio/src/dynamic-preprocessors/ssl/Makefile.am	2011-08-24 13:03:56.007091304 -0500
@@ -3,11 +3,11 @@
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_ssl_preproc.la
 
-libsf_ssl_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_ssl_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_ssl_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
