--- snort-2.9.3/configure.in.orig	2013-04-04 22:26:48.988446980 +0100
+++ snort-2.9.3/configure.in	2013-04-04 22:26:56.352446550 +0100
@@ -3,7 +3,7 @@
 AC_CONFIG_SRCDIR([src/snort.c])
 AC_PREREQ(2.50)
 #LT_PREREQ([2.2.6])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 # When changing the snort version, please also update the VERSION
 # definition in "src/win32/WIN32-Includes/config.h"
 AM_INIT_AUTOMAKE(snort,2.9.3)
