--- src/dynamic-preprocessors/modbus/Makefile.am	2012-08-08 06:23:42.022971499 +0400
+++ src/dynamic-preprocessors/modbus/Makefile.am	2012-08-08 06:24:35.994831043 +0400
@@ -3,11 +3,11 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_modbus_preproc.la
 
-libsf_modbus_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_modbus_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_modbus_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
--- src/dynamic-preprocessors/dnp3/Makefile.am	2012-04-27 22:09:56.000000000 +0400
+++ src/dynamic-preprocessors/dnp3/Makefile.am	2012-08-08 07:38:53.499356940 +0400
@@ -3,11 +3,11 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_dnp3_preproc.la
 
-libsf_dnp3_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_dnp3_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_dnp3_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
--- src/dynamic-preprocessors/gtp/Makefile.am	2012-04-27 22:09:56.000000000 +0400
+++ src/dynamic-preprocessors/gtp/Makefile.am	2012-08-08 07:40:54.313711828 +0400
@@ -3,11 +3,11 @@ AUTOMAKE_OPTIONS=foreign no-dependencies
 
 INCLUDES = -I../include -I${srcdir}/../libs -I$(srcdir)/includes
 
-libdir = ${exec_prefix}/lib/snort_dynamicpreprocessor
+libdir = ${libexecdir}/dynamicpreprocessor
 
 lib_LTLIBRARIES = libsf_gtp_preproc.la
 
-libsf_gtp_preproc_la_LDFLAGS = -export-dynamic -module @XCCFLAGS@
+libsf_gtp_preproc_la_LDFLAGS = -shared -export-dynamic -module @XCCFLAGS@ -avoid-version
 if SO_WITH_STATIC_LIB
 libsf_gtp_preproc_la_LIBADD = ../libsf_dynamic_preproc.la
 else
